scripts/publish.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Custom publishing script which wraps `npm publish` to only attempt publishing
+ * packages that have not yet been published.
+ *
+ * @packageDocumentation
+ */
+
+// @ts-check
+
+const { workspaces } = require('../package.json')
+const { glob } = require('glob')
+const path = require('node:path')
+const fs = require('node:fs/promises')
+const util = require('node:util')
+const exec = util.promisify(require('node:child_process').exec)
+const { spawn } = require('node:child_process')
```

Comment:
exec, because it's easy to promisify and we read the output as json
spawn, because we want to print the npm output to console (shows what was packed into the tarball)